geode-core/src/main/java/org/apache/geode/pdx/internal/AutoSerializableManager.java

```diff
@@ -2434,20 +2434,11 @@ public boolean equals(Object obj) {
     AutoSerializableManager other = (AutoSerializableManager) obj;
     if (checkPortability != other.checkPortability)
       return false;
-    if (classPatterns == null) {
-      if (other.classPatterns != null)
-        return false;
-    } else if (!classPatterns.equals(other.classPatterns))
+    if (!classPatterns.equals(other.classPatterns))
```

Comment:
Can be made simple by having OR condition between all 3 checks. Its up to you what style you want to follow.

Reply:
Fixed